contracts/test/integration/vip/vip13.sol

```diff
@@ -0,0 +1,484 @@
+import {ERC20CompoundPCVDeposit} from "../../../pcv/compound/ERC20CompoundPCVDeposit.sol";
+import {PriceBoundPSM} from "../../../peg/PriceBoundPSM.sol";
+import {IPegStabilityModule} from "../../../peg/IPegStabilityModule.sol";
+import {VoltMigrator} from "../../../volt/VoltMigrator.sol";
+import {VoltV2} from "../../../volt/VoltV2.sol";
+import {IVolt} from "../../../volt/IVolt.sol";
+import {IPCVDeposit} from "../../../pcv/IPCVDeposit.sol";
+import {MigratorRouter} from "../../../pcv/MigratorRouter.sol";
+
+contract vip13 is DSTest, IVIP {
+    using SafeERC20 for IERC20;
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+    uint256 public voltInUsdcPSM;
+    uint256 public voltInDaiPSM;
+
+    PriceBoundPSM public voltV2DaiPriceBoundPSM;
+    PriceBoundPSM public voltV2UsdcPriceBoundPSM;
+
+    VoltV2 public voltV2;
+
+    VoltMigrator public voltMigrator;
+    MigratorRouter public migratorRouter;
+
+    ITimelockSimulation.action[] private proposal;
+
+    uint256 voltDaiFloorPrice = 9_000;
+    uint256 voltDaiCeilingPrice = 10_000;
+
+    uint256 voltUsdcFloorPrice = 9_000e12;
+    uint256 voltUsdcCeilingPrice = 10_000e12;
+
+    /// @notice target token balance for the DAI PSM to hold
+    uint248 private constant targetBalanceDai = 100_000e18;
+
+    /// @notice target token balance for the USDC PSM to hold
+    uint248 private constant targetBalanceUsdc = 100_000e6;
+
+    /// @notice scale up USDC value by 12 decimals in order to account for decimal delta
+    /// and properly update the buffer in ERC20Allocator
+    int8 private constant usdcDecimalNormalizer = 12;
+
+    /// @notice no scaling to do on DAI as decimals are 18
+    int8 private constant daiDecimalNormalizer = 0;
+
+    ERC20Allocator private allocator =
+        ERC20Allocator(MainnetAddresses.ERC20ALLOCATOR);
+
+    ERC20CompoundPCVDeposit private daiDeposit =
+        ERC20CompoundPCVDeposit(MainnetAddresses.COMPOUND_DAI_PCV_DEPOSIT);
+    ERC20CompoundPCVDeposit private feiDeposit =
+        ERC20CompoundPCVDeposit(MainnetAddresses.COMPOUND_FEI_PCV_DEPOSIT);
+    ERC20CompoundPCVDeposit private usdcDeposit =
+        ERC20CompoundPCVDeposit(MainnetAddresses.COMPOUND_USDC_PCV_DEPOSIT);
+
+    constructor() {}
+
+    function getMainnetProposal()
+        public
+        override
+        returns (ITimelockSimulation.action[] memory prop)
+    {
+        voltV2 = new VoltV2(MainnetAddresses.CORE);
+        voltMigrator = new VoltMigrator(MainnetAddresses.CORE, address(voltV2));
+        deployPsms(address(voltV2));
+
+        migratorRouter = new MigratorRouter(
+            address(voltV2),
+            address(voltMigrator),
+            voltV2DaiPriceBoundPSM,
+            voltV2UsdcPriceBoundPSM
+        );
+
+        address[] memory toWhitelist = new address[](3);
+        toWhitelist[0] = address(voltV2UsdcPriceBoundPSM);
+        toWhitelist[1] = address(voltV2DaiPriceBoundPSM);
+        toWhitelist[2] = address(voltMigrator);
```

Comment:
we can add the migrator to the pcv guardian, however it doesn't conform to the PCV deposit interface, so in a case where we want to pull funds from it using a PCV Guard, we wouldn't be able to.

we have 2 options here:
1. make migrator contract implement the PCV deposit interface
2. not add migrator contract to the PCV Guardian list